xbmc/cores/VideoRenderers/OverlayRenderer.cpp

```diff
@@ -328,12 +328,32 @@ COverlay* CRenderer::Convert(CDVDOverlaySSA* o, double pts)
 {
   CRect src, dst;
   g_renderManager.GetVideoRect(src, dst);
-
-  int width  = MathUtils::round_int(dst.Width());
-  int height = MathUtils::round_int(dst.Height());
-
+  int videoWidth = MathUtils::round_int(dst.Width());
+  int videoHeight = MathUtils::round_int(dst.Height());
+
+  // libass render in a target area which named as frame. the frame size may bigger than video size,
+  // and including margins between video to frame edge. libass allow to render subtitles into the margins.
+  // this has been used to show subtitles in the top or bottom "black bar" between video to frame border.
+  // here use g_graphicsContext.GetViewWindow() to get the target area.
+  CRect target = g_graphicsContext.GetViewWindow();
```

Comment:
this may work but is likely to break with future changes. renderer has control over the target rendering area of video, not gfx context. why not adding a 3rd argument to g_renderManager.GetVideoRect which returns the desired region